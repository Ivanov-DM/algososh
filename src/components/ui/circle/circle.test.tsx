import React from 'react';
import renderer from "react-test-renderer";

import {Circle} from "./circle";
import {ElementStates} from "../../../types/element-states";

describe('Test for circle component', () => {
    it('Circle renders correctly without letter', () => {
        const tree = renderer
            .create(<Circle/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle renders correctly with letters', () => {
        const tree = renderer
            .create(<Circle letter={'ABC'}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle renders correctly with attribute head as string', () => {
        const tree = renderer
            .create(<Circle letter={'ABC'} head={'1'}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle renders correctly with attribute head as react element', () => {
        const tree = renderer
            .create(<Circle letter={'ABC'} head={<Circle letter={'CAB'}/>}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle renders correctly with attribute tail as string', () => {
        const tree = renderer
            .create(<Circle letter={'ABC'} tail={'1'}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle renders correctly with attribute tail as react element', () => {
        const tree = renderer
            .create(<Circle letter={'ABC'} tail={<Circle letter={'CAB'}/>}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle renders correctly with attribute index', () => {
        const tree = renderer
            .create(<Circle letter={'ABC'} index={1}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle renders correctly with props isSmall === true', () => {
        const tree = renderer
            .create(<Circle letter={'ABC'} isSmall={true}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle renders correctly with attribute state === `ElementStates.Default`', () => {
        const tree = renderer
            .create(<Circle letter={'ABC'} state={ElementStates.Default}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle renders correctly with attribute state === `ElementStates.Changing`', () => {
        const tree = renderer
            .create(<Circle letter={'ABC'} state={ElementStates.Changing}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle renders correctly with attribute state === `ElementStates.Modified`', () => {
        const tree = renderer
            .create(<Circle letter={'ABC'} state={ElementStates.Modified}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});