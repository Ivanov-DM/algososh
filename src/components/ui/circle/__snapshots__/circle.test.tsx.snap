// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for circle component Circle renders correctly with attribute head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    1
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Test for circle component Circle renders correctly with attribute head as react element 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          CAB
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Test for circle component Circle renders correctly with attribute head as string 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    1
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Test for circle component Circle renders correctly with attribute index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Test for circle component Circle renders correctly with attribute state === \`ElementStates.Changing\` 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Test for circle component Circle renders correctly with attribute state === \`ElementStates.Default\` 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Test for circle component Circle renders correctly with attribute state === \`ElementStates.Modified\` 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Test for circle component Circle renders correctly with attribute tail as react element 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          CAB
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Test for circle component Circle renders correctly with attribute tail as string 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    1
  </div>
</div>
`;

exports[`Test for circle component Circle renders correctly with letters 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Test for circle component Circle renders correctly with props isSmall === true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Test for circle component Circle renders correctly without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
